src/main/java/de/cdiag/launchpadbackend/service/UserService.java

```diff
@@ -171,4 +174,13 @@ public App updateApplication(String username, App providedApp) {
         //return the updated
         return savedApp;
     }
+
+    public User register(User user) {
+        // check if user already exists
+        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
+            throw new UserAlreadyExistsException("username '" + user.getUsername() + "' already exists.");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code contains a potential information leakage in the error handling for the following call(s): &#39;User.getUsername()&#39; (line 181). You are handling this error data through: &#39;UserAlreadyExistsException.&lt;init&gt;&#39; (line 181). There are methods available that could be added to handle sensitive data, like masking and redaction. For more information about information leakage, see https://cwe.mitre.org/data/definitions/209.html.